substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVFileRecord.java

```diff
@@ -0,0 +1,152 @@
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.objectfile.pecoff.cv;
+
+import com.oracle.objectfile.debugentry.FileEntry;
+
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+final class CVFileRecord extends CVSymbolRecord {
+
+    private static final byte CHECKSUM_NONE = 0x00;
+    private static final byte CHECKSUM_MD5 = 0x01;
+    private static final byte CB_VALUE = 0x10;
+
+    private static final int FILE_RECORD_LENGTH = 24;
+
+    private static final int CHECKSUM_LENGTH = 16;
+    private static final byte[] EMPTY_CHECKSUM = new byte[CHECKSUM_LENGTH];
+
+    private static final int FILE_TABLE_INITIAL_SIZE = 200;
+
+    private final CVSymbolSectionImpl.CVStringTable strings;
+
+    private int currentOffset = 0;
+    private Map<FileEntry, Integer> fileEntryToOffsetMap = new LinkedHashMap<>(FILE_TABLE_INITIAL_SIZE);
+
+    CVFileRecord(CVDebugInfo cvDebugInfo, CVSymbolSectionImpl.CVStringTable strings) {
+        super(cvDebugInfo, CVDebugConstants.DEBUG_S_FILECHKSMS);
+        this.strings = strings;
+    }
+
+    /**
+     * Convert a FileEntry to a path. Avoid an NPE if there's no DirEntry.
+     *
+     * @param fileEntry
+     * @return path string, including directory if it exists.
+     */
+    private static String fixPath(FileEntry fileEntry) {
+        return fileEntry.getDirEntry() != null ? fileEntry.getFullName() : fileEntry.getFileName();
+    }
+
+    int addFile(FileEntry entry) {
+        if (fileEntryToOffsetMap.containsKey(entry)) {
+            return fileEntryToOffsetMap.get(entry);
+        } else {
+            fileEntryToOffsetMap.put(entry, currentOffset);
+            /* create required stringtable entry */
+            strings.add(fixPath(entry));
+            currentOffset += FILE_RECORD_LENGTH;
+            return currentOffset - FILE_RECORD_LENGTH;
+        }
+    }
+
+    @Override
+    public int computeSize(int initialPos) {
+        /* add all fileEntries; duplicates are ignored */
```

Comment:
@stooke please fix all your comments.

- Use of `//` is discouraged.
- Comments almost always start with an uppercase letter in our codebase
- Merge consecutive comment blocks (like this one) (no `*/` immediately followed by `/*`)
If you want to prevent eclipseformat  to mess with your comment formatting start your comments with `/*-`.

Please clean up your commit log.
- Squash commits together were possible (use `git rebase --interactive origin/master`)
- Commit messages should start with an uppercase letter and should not contain spelling errors.